Framework/DataHandling/src/LoadVulcanCalFile.cpp

```diff
@@ -536,18 +537,17 @@ void LoadVulcanCalFile::convertOffsets() {
         "The simple version to calcualte detector's 2theta fails on this "
         "situation.");
 
+  const auto &spectrumInfo = m_tofOffsetsWS->spectrumInfo();
   map<int, pair<double, double>>::iterator mfiter;
   for (size_t iws = 0; iws < numspec; ++iws) {
     // Get detector's information including bank belonged to and geometry
     // parameters
-    Geometry::IDetector_const_sptr det = m_tofOffsetsWS->getDetector(iws);
-    V3D detPos = det->getPos();
 
-    detid_t detid = det->getID();
+    detid_t detid = spectrumInfo.detector(iws).getID();
     int bankid = detid / static_cast<int>(NUMBERRESERVEDPERMODULE);
 
     double l2, twotheta, phi;
-    detPos.getSpherical(l2, twotheta, phi);
+	spectrumInfo.position(iws).getSpherical(l2, twotheta, phi);
```

Comment:
You can then probably also remove the call to `getInstrumentParameters` and just use `SpectrumInfo::l1`.